lib/Mongo/MongoCollection.php

```diff
@@ -473,15 +535,30 @@ public function findOne(array $query = [], array $fields = [], array $options =
      */
     public function createIndex(array $keys, array $options = [])
     {
-        // Note: this is what the result array should look like
-//        $expected = [
-//            'createdCollectionAutomatically' => true,
-//            'numIndexesBefore' => 1,
-//            'numIndexesAfter' => 2,
-//            'ok' => 1.0
-//        ];
+        if (! is_array($keys) || ! count($keys)) {
+            throw new MongoException('keys cannot be empty');
+        }
+
+        // duplicate
+        $indexes = $this->collection->listIndexes();
+        foreach ($indexes as $index) {
+            if ($index->getKey() === $keys) {
+                throw new \MongoDuplicateKeyException();
```

Comment:
Regarding, this the code is still throwing `MongoDuplicateKeyException` rather than `MongoResultException`. I kept that way because I don't know if duplicate exception would be ever use and in exactly what situation. If you have the chance to look at the original driver code and tell me. Worst case scenario, I can take a look on my own tomorrow night.